lib/src/main/kotlin/com/swmansion/starknet/crypto/Blake.kt

```diff
@@ -0,0 +1,149 @@
+                }
+                unpackedU32s[start] = unpackedU32s[start] or bigMarker
+            }
+        }
+        return unpackedU32s
+    }
+
+    /**
+     * Packs the first 32 bytes of a byte array (little-endian) into a Felt.
+     *
+     * @param hashBytes Byte array containing the hash.
+     * @return Felt representation of the packed value.
+     */
+    private fun pack256LeToFelt(hashBytes: ByteArray): Felt {
+        require(hashBytes.size >= 32) { "need at least 32 bytes to pack" }
+
+        var value = BigInteger.ZERO
+        for (i in 31 downTo 0) {
+            value = value.shiftLeft(8).add(BigInteger.valueOf((hashBytes[i].toInt() and 0xFF).toLong()))
+        }
+        return value.mod(Felt.PRIME).toFelt
+    }
+
+    /**
+     * Computes the Blake2s hash of the given data and converts it to a Felt.
+     *
+     * @param data Byte array to hash.
+     * @return Felt representation of the hash.
+     */
+    private fun blake2sToFelt(data: ByteArray): Felt {
+        val digest = MessageDigest.getInstance("BLAKE2S-256", "BC")
+        val hash = digest.digest(data)
+        return pack256LeToFelt(hash)
+    }
+
+    /**
+     * Computes the Blake2s hash of a list of Felts.
+     *
+     * @param felts List of Felt values to hash.
+     * @return Felt representation of the hash.
+     */
+    @JvmStatic
+    fun blake2sHash(felts: List<Felt>): Felt {
+        val u32Words = encodeFeltsToU32s(felts)
+
+        // Serialize as little-endian bytes
+        val byteStream = u32Words.flatMap { word ->
+            val bytes = ByteArray(4)
+            for (i in 0 until 4) {
+                bytes[i] = ((word shr (8 * i)) and 0xFF).toByte()
+            }
+            bytes.asList()
+        }.toByteArray()
+
+        return blake2sToFelt(byteStream)
+    }
+
+    /**
+     * Converts a Felt to its native byte array representation (32 bytes, big-endian).
+     *
+     * @param felt Felt value to convert.
+     * @return Byte array representation of the Felt.
+     */
+    @JvmStatic
+    fun blake2sHash(first: ByteArray, second: ByteArray): ByteArray {
+        val digest = MessageDigest.getInstance("BLAKE2S-256", "BC")
+        digest.update(first)
+        return digest.digest(second)
+    }
+
+    /**
+     * Converts pair of Felts to their native byte array representations and computes their Blake2s hash.
+     *
+     * @param first Felt value to convert.
+     * @param second Felt value to convert.
+     * @return Byte array representation of the hash.
+     */
+    @JvmStatic
+    fun blake2sHash(first: Felt, second: Felt): Felt {
+        val hash = blake2sHash(feltToNative(first), feltToNative(second))
+        return pack256LeToFelt(hash)
+    }
+
+    /**
+     * Converts an iterable of Felts to their native byte array representations and computes their Blake2s hash.
+     *
+     * @param values Iterable of Felt values to convert.
+     * @return Felt representation of the hash.
+     */
```

Comment:
I think docs of both of these should just say it computes the hash of 2 felts. Conversions are an implementation detail